CMakeLists.txt

```diff
@@ -71,7 +71,7 @@ set(CMAKE_INSTALL_RPATH ${IRODS_EXTERNALS_FULLPATH_CLANG_RUNTIME}/lib)
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++ -Wl,--export-dynamic -pthread") # export-dynamic so stacktrace entries from executables have function names
 set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -stdlib=libc++ -Wl,-z,defs -pthread")
 set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++ -Wl,-z,defs -pthread")
-add_compile_options(-nostdinc++ -Wall -Wextra -Werror -Wno-unused-function -Wno-unused-parameter)
+add_compile_options(-nostdinc++ -Wall -Wextra -Werror -Wno-unused-function -Wno-unused-parameter -Wno-unknown-warning-option -Wno-error=deprecated-copy -Wno-error=tautological-overlap-compare)
```

Comment:
From the commit message:
```
-Wno-unknown-warning-option:
    Stop clang 6 from whining about new warning options.
-Wno-error=deprecated-copy
    Do not error on definition of implicit copy constructor for
    classes with a user-declared copy assignment operator.
    This is needed because the boost headers throw this warning.
-Wno-error=tautological-overlap-compare
    Do not error when overlapping comparisons always evaluate to
    false.
    This is needed because the boost headers throw this warning.
```